### PR TITLE
INT-7144 - Check for duplicate keys

### PR DESCRIPTION
**Pull request by RonaldEAM**

### CODE REVIEW

File: src/steps/repositories.ts [hunk 1]
```diff
@@ -167,13 +167,15 @@ export async function fetchArtifacts({
...
-                to: artifactEntity,
-              }),
-            );
+            const repoArtifactRelationship = createDirectRelationship({
+              _class: RelationshipClass.HAS,
+              from: repositoryEntity,
+              to: artifactEntity,
+            });
+
+            if (!(await jobState.hasKey(repoArtifactRelationship._key))) {
```
Comment (VDubber) [inline src/steps/repositories.ts:176]: were you able to verify that they were true duplicates? Did the logs indicate that properties were all 100% the same?

File: src/steps/repositories.ts [hunk 2]
```diff
@@ -167,13 +167,15 @@ export async function fetchArtifacts({
...
+              await jobState.addRelationship(repoArtifactRelationship);
+            }
```
Comment (dobregons) [inline src/steps/repositories.ts:178]: Shouldn't we `logger.warn` when this case occurs (else)?

File: src/steps/repositories.ts (hunk 1 above)
Comment (RonaldEAM) [reply src/steps/repositories.ts:176]: No report is generated for duplicate relationships, only for duplicate entities. This is the error:
```
Duplicate _key detected (_key=artifactory_repository:docker-local|has|artifactory_artifact:https://a0us.jfrog.io/artifactory/docker-local/docker/managed-marketplace/PR-155-1-8170e70-dev-linux-amd64/sha256__12eae9349b7b133b0fa7d6e7e24024d2ab71f2fae9a3e14d243494e29a3bc30b)
```
I don't think I repo could have the same path/filename, I'm just guessing artifactory's pagination has some overlapping of records

Comment (RonaldEAM) [reply src/steps/repositories.ts:176]: Perhaps for now I can log a report manually as David mentioned so we can check what's the difference without the step failing?

Comment (VDubber) [reply src/steps/repositories.ts:176]: Yeah, sounds good. Let's get logs and then monitor what we see. Does the relationship have and customer data properties or just the key? We don't want to log any sensitive data.

Comment (VDubber) [reply src/steps/repositories.ts:176]: Actually, if the relationship does not have properties, we should be safe to ignore and not log.

Comment (RonaldEAM) [reply src/steps/repositories.ts:165]: yeah I meant logging the duplicated entity, that's the source of the issue, I added that in my commit, you can check it

**Review by VDubber — APPROVED** [review]

**Pull request merged**